### PR TITLE
Addresses BACKLOG-40289: add support for the necessary XMLA properties and rowset

### PR DESCRIPTION
**Pull request by mbrasil**
@pentaho/millenniumfalcon please review

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 25m 9s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
:ok_hand: All tests passed!
 
Tests run: **3307**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/mondrian/job/PR-1374/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 27m 29s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 23m 19s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 20m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 20m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 21m 6s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 19m 55s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 22m 22s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
:ok_hand: All tests passed!
 
Tests run: **3307**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/mondrian/job/PR-1374/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by afrjorge — APPROVED** [review]

File: mondrian/src/main/java/mondrian/xmla/XmlaHandler.java [hunk 1]
```diff
@@ -842,6 +954,8 @@ private QueryResult executeQuery( XmlaRequest request ) throws XmlaException {
       } catch ( Exception ex ) {
         throw new XmlaException( SERVER_FAULT_FC, HSB_EXECUTE_QUERY_CODE, HSB_EXECUTE_QUERY_FAULT_FS, ex );
       }
+    } catch ( OlapException e ) {
+      throw new RuntimeException( e );
```
Comment (loonatic) [inline mondrian/src/main/java/mondrian/xmla/XmlaHandler.java:958]: Instead directly throwing `RuntimeException`, maybe throw `MondrianException` (or other that extends RE) to allow more flexibility?

**Review by loonatic — APPROVED** [review]
Overall looks good, I left two comments but one to double-check and other is up to you.

There are also a few mentions to `ArrayList<..> someVar` instead of `List<..> someVar` but again, small thing.

**Review by afrjorge — APPROVED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 31m 40s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
:ok_hand: All tests passed!
 
Tests run: **3307**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/mondrian/job/PR-1374/9/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 10s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
:ok_hand: All tests passed!
 
Tests run: **3307**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/mondrian/job/PR-1374/10/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 18m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 18m 4s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
Command "git checkout -f b27d894695e0e91f3aed0fd68f8aec004f445871" returned status code 128:
stdout: 
stderr: fatal: reference is not a tree: b27d894695e0e91f3aed0fd68f8aec004f445871
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 46s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
Command "git checkout -f 2ded88303c2b95f5e99ee30088a3e46650889ed8" returned status code 128:
stdout: 
stderr: fatal: reference is not a tree: 2ded88303c2b95f5e99ee30088a3e46650889ed8
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 34s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
Command "git checkout -f 2ded88303c2b95f5e99ee30088a3e46650889ed8" returned status code 128:
stdout: 
stderr: fatal: reference is not a tree: 2ded88303c2b95f5e99ee30088a3e46650889ed8
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 22m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -DrunITs -pl mondrian
```
---
:ok_hand: All tests passed!
 
Tests run: **3307**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/mondrian/job/PR-1374/16/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**